main/res/values/strings.xml

```diff
@@ -252,6 +253,8 @@
     <string name="info_stored_image">New image saved to:</string>
     <string name="info_storing_static_maps">Trying to store static maps</string>
     <string name="info_cache_saved">The cache has been stored locally</string>
+    <string name="info_calculator_state_saved">Current state of the calculator has been saved to waypoint.</string>
```

Comment:
> Keep toasts for things that aren't expected.

But that's my point, it _is_ unexpected!

> "The calculation result was invalid, no coordinates could be set but the current state of your calculation has been saved to the waypoint"

At the moment the messages are displayed as two separate toasts.
When the coordinates _are_ valid it shows:
> Current state of the calculator has been saved to waypoint.

> Be sure to save waypoint if you want to keep the calculator state.

And when the coordinates _aren't_ valid it shows:
> Current state of the calculator has been saved to waypoint.

> c:geo can\'t parse latitude or longitude.

And honestly, it's perfectly fine.

I acknowledge it's somewhat in-elegant to have two toasts display one after the other but it actually plays out perfectly well.  As I said before, I've put a lot of thought into this and have tried several approaches and this is the best I've been able to find and I recommend giving it a go.  If we need to change it we'll change it but it actually works perfectly fine as it is.